src/components/fields/ArrayField.js

```diff
@@ -44,7 +46,10 @@ function DefaultArrayItem(props) {
     fontWeight: "bold",
   };
   return (
-    <div key={props.index} className={props.className}>
+    <div
+      key={props.key}
+      id={`array-item-${props.key}`}
```

Comment:
I don't think there's any point in exposing the key to the user as the `id`? Currently ids are only used as per the idSchema (for example, `root_listOfStrings_0`) so having another function for id may also not be the best.

Reply:
What would you recommend here?

Reply:
Can we just remove the id?

Or is there a particular reason why you wanted to add it?

Reply:
I'm using the id right now to ensure that the key does not change inside the ArrayField tests. If you remove the id, the tests for inserting new rows will fail. The re-order and delete tests only pass when the attribute is removed because the id both before and after the action are null, but these should really test to ensure an id exists as well.

If we remove the id, I'm not sure how these tests should be updated to verify the correct behavior as the key doesn't appear to be exposed anywhere else.

Regarding the `idSchema`, were you thinking of just passing the key into the `toIdSchema` method for each item, and then accessing it via `props.idSchema.key`?

Reply:
Got it. I don't think we should add it to the idSchema or expose it as the id attribute, though, given that the array item ids have no meaning apart from being unique and stable (as opposed to the other ids in the idSchema).

Here are two possible solutions:
- Can we expose the key as `data-rjsf-id` or something instead?
- Or, we don't expose the key at all -- in our tests, create a custom ArrayFieldTemplate that exposes the key in the DOM so that the tests can access it.

Reply:
I removed the id attribute, and updated the tests to use a custom template.